rocAL/source/pipeline/tensor.cpp

```diff
@@ -201,6 +211,42 @@ void Tensor::update_tensor_roi(const std::vector<uint32_t> &width,
                 roi[i].xywh.h = height[i];
             }
         }
+    } else if (!_info.is_metadata()) {  // Audio - Data
+        auto max_dims = _info.max_shape();
+        unsigned max_samples = max_dims.at(0);
+        unsigned max_channels = max_dims.at(1);
+        auto samples = width;
+        auto channels = height;
+
+        if (samples.size() != channels.size())
```

Comment:
Not sure what this logic is doing. We need to discuss

Reply:
Similar to updating the Tensor's ROI with height and width in case of images
For audio the samples and channels are updated in the Tensor's ROI